tests/unit/test_testing/test_lifespan_handler.py

```diff
@@ -35,3 +38,17 @@ async def app(scope: Scope, receive: Receive, send: Send) -> None:
         handler = LifeSpanHandler(TestClient(app))
         await handler.wait_shutdown()
         handler.close()
+
+
+async def test_multiple_clients_event_loop() -> None:
+    @get("/")
+    def return_loop_id() -> dict:
+        return {"loop_id": id(get_event_loop())}
+
+    app = Litestar(route_handlers=[return_loop_id])
+
+    async with AsyncTestClient(app) as client_1, AsyncTestClient(app) as client_2:
```

Comment:
This should even be the same loop with different client instances, right?

Reply:
i think so, this added test is just the mcve from the issue and it fails before the changes, is there something I should add ?

Reply:
I realize the diff is overly complicate, the pr is simple: before AsyncTestClient used TestClientTransport hence the issue.
so I created a BaseClientTransport that essentially takes care of request/response and added a ASGITestClientTransport that inherits httpx ASGIBaseTransport and ovverrides handle_async_request while TestClientTransport overrides handle_request, the BaseClientTransport is what's common between te 2 transports.

Reply:
> i think so, this added test is just the mcve from the issue and it fails before the changes, is there something I should add ?

Yeah, maybe a test for the transport in particular that shows the client always uses the currently running loop?

Reply:
actually if this test was using TestClient instead of AsyncTestClient it wouldn't pass as `start_blocking_portal` creates a new event loop.

> This should even be the same loop with different client instances, right?

by different client instances you meant a AsyncTestClient and a TestClient ?

Reply:
No I mean different instances `AsyncClient`. But actually, I think the test should probably just show that the client uses the running loop from the context it's created in.